--- linux-3.4.6/fs/ext2/file.c  2012-07-20 00:41:49.000000000 +0530
+++ ../file.c   2012-08-03 17:05:20.275017854 +0530
@@ -45,7 +45,7 @@
int ret;
struct super_block *sb = file->f_mapping->host->i_sb;
struct address_space *mapping = sb->s_bdev->bd_inode->i_mapping;
-
+   printk (KERN_INFO "linuxKernel101 : ext2_fsync called");
ret = generic_file_fsync(file, start, end, datasync);
if (ret == -EIO || test_and_clear_bit(AS_EIO, &mapping->flags)) {
        /* We don't really know where the IO error happened... */

